trinity/plugins/builtin/tx_pool/plugin.py

```diff
@@ -61,7 +61,10 @@ def configure_parser(self, arg_parser: ArgumentParser, subparser: _SubParsersAct
 
     def handle_event(self, activation_event: BaseEvent) -> None:
         if isinstance(activation_event, TrinityStartupEvent):
-            self.is_enabled = activation_event.args.tx_pool
+            light_mode = activation_event.args.sync_mode == 'light'
+            self.is_enabled = activation_event.args.tx_pool and not light_mode
```

Comment:
Mmh, I feel different about that.

>Did you consider changing trinity's `launch_node()` to modify `args.tx_pool` when in light mode

Since that code is currently in the core Trinity code base it should not know about the transaction pool plugin. Basically, none of the Trinity core code should know about any plugins imho.

On the other hand, I feel less bad about *this particular* transaction pool plugin making assumptions of the `sync_mode`. After all, the whole idea is that someone else could write another transaction pool plugin which works just fine with light mode.

Reply:
Hmm, indeed, I guess it makes more sense to have the plugins themselves decide when they should be enabled. In this case my only suggestion is to use a constant when comparing to `.args.sync_mode`. I believe we might even have that already: `trinity.consntants.SYNC_LIGHT`

Reply:
Ah, good point. Will update to use that constant.